examples/custom_generic_nl_family_custom_types.rs

```diff
@@ -0,0 +1,140 @@
+///
+/// THIS EXAMPLE WILL NOT WORK ON YOUR SYSTEM UNLESS YOU CODE AND LOAD A CUSTOM KERNEL MODULE
+/// THAT IMPLEMENTS THIS BASIC (GENERIC) NETLINK FUNCTIONALITY. SEE ALSO HERE:
+/// https://github.com/phip1611/generic-netlink-user-kernel-rust
+fn main() {
+    // this gets registered by your custom Linux kernel module/driver
+    let family_name = "your_family_name_here";
+    // connect
+    let mut sock = NlSocketHandle::connect(
+        NlFamily::Generic,
+        // target pid is 0 => kernel (we send messages to kernel)
+        Some(0),
+        U32Bitmask::empty(),
+    )
+    .unwrap();
+    // family_name to family_id (temporarily assigned netlink family Id by kernel, after registering the generic netlink family)
+    let family_id = sock
+        .resolve_genl_family(family_name)
+        .expect("Kernel Module that registered custom family must exist!");
+    println!(
+        "Netlink family id of generic netlink family with name '{}' is: {}",
+        family_name, family_id
+    );
+
+    // We want to send an Echo command
+    // 1) prepare Foobar Attribute
+    let mut attrs: GenlBuffer<FoobarAttribute, Buffer> = GenlBuffer::new();
+    attrs.push(
+        Nlattr::new(
+            None,
+            false,
+            false,
+            FoobarAttribute::Msg,
+            "Some data that has `Nl` trait implemented, like &str",
+        )
+        .unwrap(),
+    );
+    // 2) prepare Generic Netlink Header. The Generic Netlink Header contains the
+    //    attributes (actual data) as payload.
+    let gnmsghdr = Genlmsghdr::new(
+        FoobarOperation::Echo,
+        // Version is dependent on your kernel module
+        // you can check it there and make special action, or you ignore it
+        1,
+        attrs,
+    );
+    // 3) Prepare Netlink header. The Netlink header contains the Generic Netlink header
+    //    as payload.
+    let nlmsghdr = Nlmsghdr::new(
+        None,
+        family_id,
+        // This depends on your kernel module. Do you check there if any flags are used?
+        // Request is required (TODO by neli or by netlink?)
+        // others are up to you
+        NlmFFlags::new(&[NlmF::Request]),
+        None,
+        // Port ID. Not necessarily the process id of the current process. This field
+        // could be used to identify different points or threads inside your application
+        // that send data to the kernel. This has nothing to do with "routing" the packet to
+        // the kernel
+        Some(process::id()),
+        NlPayload::Payload(gnmsghdr),
+    );
+
+    // Send data
+    sock.send(nlmsghdr).expect("Send must work");
+
+    // receive echo'ed message
+    let res: Nlmsghdr<u16, Genlmsghdr<FoobarOperation, FoobarAttribute>> =
+        sock.recv().expect("Should receive a message").unwrap();
+
+    // According to Netlink Linux kernel code nl_type is either 0x2 (error) or our family id (ok)
+    if res.nl_type == 0x2 {
```

Comment:
Ah I have to refactor this. Looks like this check is already covered by sock.recv(). It returns Err if nl_type is 0x2

Reply:
No, neli tells `The buffer was not large enough to complete the deserialize operation`. Doesn't look what I would neli expect to do. Technically an "well formed error message" is okay but not error